figma-shop/src/app/shared/services/product.service.ts

```diff
@@ -19,6 +20,12 @@ export class ProductServiceService {
   }
 
   public getAllProducts(): Promise<ProductInterface> {
```

Comment:
are you sure that `getAllProducts()` returns `Promise<ProductInterface>`, not `Promise<ProductInterface[]>`?

Reply:
my mistake, thanks